debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -166,34 +161,53 @@ public String connectionString() {
                 + "Make sure there are no long-running transactions running in parallel as they may hinder the allocation of the replication slot when starting this connector");
     }
 
+    protected ServerInfo.ReplicationSlot queryForSlot(String slotName, String database, String pluginName,
+                                                      ResultSetMapper<ServerInfo.ReplicationSlot> map) throws SQLException {
+        return prepareQueryAndMap("select * from pg_replication_slots where slot_name = ? and database = ? and plugin = ?", statement -> {
+            statement.setString(1, slotName);
+            statement.setString(2, database);
+            statement.setString(3, pluginName);
+        }, map);
+    }
+
     private Long parseConfirmedFlushLsn(String slotName, String pluginName, String database, ResultSet rs) {
         Long confirmedFlushedLsn = null;
 
         try {
-            String confirmedFlushLSNString = rs.getString("confirmed_flush_lsn");
-            if (confirmedFlushLSNString == null) {
-                return null;
-            }
+            confirmedFlushedLsn = tryParseLsn(slotName, pluginName, database, rs, "confirmed_flush_lsn");
+        } catch (SQLException e) {
```

Comment:
Minor nit on style: can you please move any code after a "}" to the next line (here and a few other places)? While not consistently enforced acros the code base, it's the style we want to converge to eventually. So let's avoid adding new code using the old style. Thanks!